File: cpp/gfg_160/pushZerosToEnd.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

class Solution {
  public:
    void pushZerosToEnd(vector<int> &arr) {
      int nonzero_index = 0;
      int n = arr.size();

      for (size_t i = 0; i < n; i++)
      {
        if (arr[i] != 0) {
          int temp = arr[i];
          arr[i] = arr[nonzero_index];
          arr[nonzero_index] = temp;
          nonzero_index++;
        }
      }
    }
};

int main() {
  int t;
  cin >> t;
  cin.ignore();

  while (t--) {
    vector<int> arr;
    string input;
    getline(cin, input);
    stringstream ss(input);
    int number;

    while (ss >> number) 
    {
      arr.push_back(number);
    }
    Solution obj;
    int n = arr.size();
    obj.pushZerosToEnd(arr);
    for (size_t i = 0; i < n; i++)
    {
      cout << arr[i] << " ";
    }    
    cout << "\n" << endl;
  }
  return 0;
}

File: python/gfg_160/pushZerosToEnd.py
class Solution:
  # Method 1
  '''
  def pushZerosToEnd(self, arr):   
    n = len(arr)
    nonzero_index = 0
    
    for i in range(n):
      if arr[i] != 0:
        arr[nonzero_index] = arr[i]
        nonzero_index += 1
            
    for i in range(nonzero_index, n):
      arr[i] = 0
'''
  # Method 2
  def pushZerosToEnd(self, arr):
    count = 0

    for i in range(len(arr)):
      if arr[i] != 0:
        arr[i], arr[count] = arr[count], arr[i]
        count += 1
    
if __name__ == "__main__":
  t = int(input())
  for _ in range(t):
    arr = list(map(int, input().split()))
    ob = Solution()
    ob.pushZerosToEnd(arr)
    n = len(arr)
    for num in arr:
      print(num, end=" ")
    print()
    t -= 1